loadgen/contract/agent-prepare-loadgen.js

```diff
@@ -48,11 +48,13 @@ const makePurseFinder = ({ wallet, walletAdmin }) => {
 
   return harden({
     /**
+     * @template {boolean} [T=false]
      * @param {Object} param0
      * @param {string} param0.brandPetname
-     * @param {boolean} [param0.existingOnly]
+     * @param {T} [param0.existingOnly]
+     * @returns {Promise<{kit: Promise<import('../types.js').NatAssetKit>, balance: Amount<'nat'>} | (true extends T ? {kit: undefined, balance: undefined} : never)>}
```

Comment:
neat

Reply:
Yeah I wish there was a better way to write these kind of overloads. This feels yucky but it worked.